### PR TITLE
Tool call result summary message

### PR DESCRIPTION
**Pull request by jspv**
## Why are these changes needed?

Creates a ToolCallResultSummaryMessage type more clearly differentiate when an agent is responding with a tool summary
vs. a TextMessage response from the LLM.  

## Related issue number

Closes [#4754](https://github.com/microsoft/autogen/issues/4754)

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by husseinmozannar — CHANGES_REQUESTED** [review]
- We recently updated the naming of toolcallmessage and toolcallresultmessage, look to pull from main and then add your changes, there is no real conflict
- The naming "ToolCallResultSummaryMessage" is a bit too long, I would just do "ToolCallSummaryMessage" perhaps, i'm sure @ekzhu has opinions

**Comment by jspv** [comment]
Code updated per request

**Comment by husseinmozannar** [comment]
Thanks!

See the failing tests, I believe ChatMessage should contain ToolCallSummaryMessage because of the Response type

**Comment by husseinmozannar** [comment]
@victordibia  This is related to #4725  I wonder what you think we should do

**Review by ekzhu — CHANGES_REQUESTED** [review]
@jspv  you can read about how to run tests in CONTRIBUTING.md

In some group chat orchestrator classes, you need to update the handling of message thread:

https://github.com/microsoft/autogen/blob/c74c2d6767cbb283e043d0c4a7fe7a3424bd89aa/python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py#L426-L427

https://github.com/microsoft/autogen/blob/c74c2d6767cbb283e043d0c4a7fe7a3424bd89aa/python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_selector_group_chat.py#L103-L104

Bonus: would be great to add test code to verify the text content has been added to the prompt that gets sent to the monkey patched model client, for SelectorGroupChat. 

Also declare message types for assistant agent, only when there are tools.

https://github.com/microsoft/autogen/blob/c74c2d6767cbb283e043d0c4a7fe7a3424bd89aa/python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py#L281-L282

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py [hunk 1]
```diff
@@ -427,7 +428,7 @@ def _thread_to_context(self) -> List[LLMMessage]:
         """Convert the message thread to a context for the model."""
         context: List[LLMMessage] = []
         for m in self._message_thread:
-            if isinstance(m, ToolCallRequestEvent | ToolCallExecutionEvent):
+            if isinstance(m, ToolCallRequestEvent | ToolCallExecutionEvent | ToolCallSummaryMessage):
```
Comment (ekzhu) [inline python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py:431]: Actually we should not ignore tool call summary messages -- they should be handled just like text message etc.

**Comment by jspv** [comment]
Thanks for the feedback.  Additions made and test cases updated, all tests passing locally.  I did not do the 'Bonus' for SelectorGroupChat, I didn't see where the current tests affected the prompt; I could add test case to cause a ToolCallSummaryMessage to be produced, but I didn't see any other Message types being actively tested in the selector_group_chat tests so decided not to stray down that path.

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py (hunk 1 above)
Comment (jspv) [reply python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_orchestrator.py:431]: Fixed.  I took my lead from the `#ignore tool call messages` comment.   Addressing now.

File: python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_selector_group_chat.py [hunk 2]
```diff
@@ -95,7 +96,7 @@ async def select_speaker(self, thread: List[AgentEvent | ChatMessage]) -> str:
         # Construct the history of the conversation.
         history_messages: List[str] = []
         for msg in thread:
-            if isinstance(msg, ToolCallRequestEvent | ToolCallExecutionEvent):
+            if isinstance(msg, ToolCallRequestEvent | ToolCallExecutionEvent | ToolCallSummaryMessage):
```
[the remark this answers is not in the document]
Comment (jspv) [reply python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_selector_group_chat.py]: done

**Review by ekzhu — APPROVED** [review]

**Comment by ekzhu** [comment]
@husseinmozannar I believe the recent changes to this PR has already resolved your requests.

**Review by husseinmozannar — APPROVED** [review]

**Review by husseinmozannar — CHANGES_REQUESTED** [review]
@victordibia I believe this will break connection.py in AGS, specifically line 328 we should add ToolCallSummary

**Comment by ekzhu** [comment]
@husseinmozannar @victordibia please take over this PR from now on to address any changes necessary

**Comment by victordibia** [comment]
I have taken a look .. I'll address any changes needed in a seperate PR. (mostly to add checks for ToolCallSummaryMessage types). Its not exactly breaking for now (those messages may not show up in the UI)

**Review by victordibia — APPROVED** [review]

**Review by husseinmozannar — APPROVED** [review]

**Pull request merged**